src/proto/Elrond.proto

```diff
@@ -0,0 +1,36 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+syntax = "proto3";
+
+package TW.Elrond.Proto;
+option java_package = "wallet.core.jni.proto";
+
+// A transaction, typical balance transfer
+message TransactionMessage {
+    uint64      nonce = 1;
+    string      value = 2;
+    string      receiver = 3;
+    string      sender = 4;
+    uint64      gas_price = 5;
```

Comment:
Elrond decimals is 18, overflow if you use type `uint64`?

Reply:
Very good question. However we expect `gas_price` to be much, much lower than `max(uint64)` for any transaction, with respect to our network's economics. The `gas_price`, chosen by the issuer of the transaction, is per gas unit - while `gas_limit` (the number of gas units) is constrained to have a quite high (~50000) lower bound, and thus a very large `gas_price` would be prohibitive (balance exhaustion). 

Once the transaction reaches the network, we use `big numbers` to perform calculations on this matter.